tyr-core/src/main/java/org/jboss/tyr/check/AddLabelCheck.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.tyr.check;
+
+import org.jboss.tyr.Check;
+import org.jboss.tyr.InvalidPayloadException;
+import org.jboss.tyr.github.GitHubService;
+import org.jboss.tyr.model.Utils;
+import org.jboss.tyr.model.yaml.LabelDetails;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonValue;
+
+@ApplicationScoped
+public class AddLabelCheck implements Check {
+
+    static final String DEFAULT_DESCRIPTION = "No description";
+    static final String DEFAULT_WHITE_COLOR = "ffffff";
+
+    @Inject
+    GitHubService gitHubService;
+
+    private String description;
+
+    private String color;
+
+    public void initCheck(LabelDetails labelDetails) {
+        if (labelDetails.getColor() != null && !labelDetails.getColor().toLowerCase().matches(Utils.COLOR_VERIFICATION)) {
+            throw new IllegalArgumentException(
+                    "Wrong color format, has to be lower character hexadecimal color code, without the leading #.");
+        }
+        this.description = labelDetails.getDescription() != null ? labelDetails.getDescription() : DEFAULT_DESCRIPTION;
+        this.color = labelDetails.getColor() != null ? labelDetails.getColor().toLowerCase() : DEFAULT_WHITE_COLOR;
```

Comment:
`TemplateChecker` will only call this method if `format.yaml` defines `label.targetFormat`. That means that the default values of `description` and `color` would not be set if that's missing. Either `initiCheck` needs to be called every time, or those fields should be initiated with the default values.